pom.xml

```diff
@@ -1771,5 +1772,49 @@
                 </plugins>
             </build>
         </profile>
+
+
+
+        <!-- JaCoCo test coverage. Will set ${surefire.jacoco.args} to be used in WildFly Test Runner config and as surefire jvm argument. -->
+        <!--      This profile generate .exec files only. This profile doesn't generate html coverage report. -->
+        <profile>
+            <id>ts.jacoco.profile</id>
+            <activation><property><name>coverage</name></property></activation>
+            <build>
+                <plugins>
+                    <plugin>
+                        <groupId>org.jacoco</groupId>
+                        <artifactId>jacoco-maven-plugin</artifactId>
+                        <version>${version.jacoco.plugin}</version>
+                        <executions>
+                            <execution>
+                                <id>ts.jacoco-prepare</id>
+                                <phase>process-test-classes</phase>
+                                <goals><goal>prepare-agent</goal></goals>
+                                <configuration>
+                                    <append>true</append>
+                                    <destFile>target/jacoco.exec</destFile>
+                                    <includes>
+                                        <include>*</include>
+                                    </includes>
+                                    <excludes>
+                                        <exclude>org.jboss.as.test.*</exclude>
+                                    </excludes>
+                                    <output>file</output>
+                                    <propertyName>surefire.jacoco.args</propertyName>
+                                </configuration>
+                            </execution>
+                        </executions>
+                    </plugin>
+                </plugins>
+            </build>
+        </profile>
+        <profile> <!-- surefire.jacoco.args property needs to be set correctly, if coverage profile is not enabled -->
+            <id>ts.jacoco.profile.not.enabled</id>
```

Comment:
you can probably remove this profile completely.
just add <surefire.jacoco.args> to top of pom.xml, similarly as we do add  <modular.jdk.args />

Reply:
If surefire.jacoco.args is defined on the top of pom.xml, jacoco-maven-plugin is not able to replace its value. I check it with <surefire.jacoco.args /> and with <surefire.jacoco.args ></surefire.jacoco.args >

Append property of jacoco-maven-plugin ( https://github.com/wildfly/wildfly-core/pull/2736/files#diff-600376dffeb79835ede4a0b285078036R1795 )is not related with property content. Append=true means, that jacoco adds data to exec, but doesn't replace exec file, see http://www.eclemma.org/jacoco/trunk/doc/prepare-agent-mojo.html